pyrasterframes/src/main/python/docs/nodata-handling.pymd

```diff
@@ -132,29 +135,134 @@ masked.select(rf_no_data_cells('blue_masked'), rf_tile_sum('mask')).show(10)
+
+One way to mask a tile is to make a new tile with a user defined NoData value. We will explore this method below. First, lets create a rasterframe from a tile with values of 0, 1, 2, and 3.
+
+```python create_dummy_tile
+tile_size = 100
+x = np.zeros((tile_size, tile_size), dtype='int16')
+
+for i in range(4):
+    x[:, i*tile_size//4:(i+1)*tile_size//4] = i
+x = Tile(x)
+
+rf = spark.createDataFrame([Row(tile=x)])
+display(x)
+```
+
+First, we mask the value of 1 by making a new column with the user defined cell type 'uint16ud1'. Then, we mask out the value of two by making a tile with the cell type 'uint16ud2'.
 
-local algebra example; same celltype what happens to nodata 
-Possibly use st_geomFromWkt and rf_rasterize to create something to work from
+```python
+def get_nodata_ct(nd_val):
+    return CellType('uint16').with_no_data_value(nd_val).cell_type_name
 
-agg
+masked_rf = rf.withColumn('tile_nd_1', 
+                           rf_convert_cell_type('tile', get_nodata_ct(1))) \
+              .withColumn('tile_nd_2',
+                          rf_convert_cell_type('tile_nd_1', get_nodata_ct(2))) \
+```
 
+```python
+collected = masked_rf.collect()
+```
 
-## Dealing with Multiple Cell Types
+Let's look at the new Tiles we created. The tile named `tile_nd_1` has the 1 values masked out as expected. 
 
-Quick demo of one ND tile one raw tile
+```python
+display(collected[0].tile_nd_1)
+```
 
-Quick demo of ND in two different cell types
+And the tile named `tile_nd_2` has the values of 1 and 2 masked out. This is because we created the tile by setting a new user defined NoData value to `tile_nd_1` the values previously masked out in `tile_nd_1` stayed masked when creating `tile_nd_2`.
+
+```python
+display(collected[0].tile_nd_2)
+```
+
+## Nodata Values in Aggregation
+
+Let's use the same tile as before to demonstrate how NoData values affect tile aggregations
+
+```python
+tile_size = 100
+x = np.zeros((tile_size, tile_size), dtype='int16')
+for i in range(4):
+    x[:, i*tile_size//4:(i+1)*tile_size//4] = i
+x = Tile(x)
+
+rf = spark.createDataFrame([Row(tile=x)])
+display(x)
+```
+
+First we create the two new masked tile columns as before. One with only the value of 1 masked, and the other with and values of 1 and 2 masked. 
+
+```python
+def get_nodata_ct(nd_val):
+    return CellType('uint16').with_no_data_value(nd_val).cell_type_name
+
+masked_rf = rf.withColumn('tile_nd_1', 
+                           rf_convert_cell_type('tile', get_nodata_ct(1))) \
+              .withColumn('tile_nd_2',
+                          rf_convert_cell_type('tile_nd_1', get_nodata_ct(2)))
+```
 
+The results of `rf_tile_sum` vary on the tiles that were masked. This is because any cells with NoData values are ignored in the aggregation. Note that `tile_nd_2` has the lowest sum, since it has the fewest amount of data cells.
 
+```python
+masked_rf.select(rf_tile_sum('tile'), rf_tile_sum('tile_nd_1'), rf_tile_sum('tile_nd_2')).show()
+```
```

Comment:
It would be worthwhile to perhaps add a level 3 header `### ` differentiating the Tile Aggregate, DataFrame Aggregate, and Local Aggregate behaviors. Here you have done Tile Agg, could expand to include the other 2 or just point to the aggreagation.md docs page 

Another thought here it may be worth working through the artithmetic for one of the cases to show the reader how to work out the expected sum. it's a block of 25*100 * value for each, right.

FWIW also there is a new section we could add  about ND in local algebra; the bug https://github.com/locationtech/rasterframes/issues/229 is now fixed to rf_local_add(t1, t2) will work like d + nd = nd.

Reply:
The code in that comment can be used as the example in the documentation.

Reply:
@vpipkt I added back the section with the local sum and NoData. I'd like you to check to make sure that sum is working correctly.